src/main/kotlin/org/opensearch/replication/action/pause/TransportPauseIndexReplicationAction.kt

```diff
@@ -71,9 +71,9 @@ class TransportPauseIndexReplicationAction @Inject constructor(transportService:
                 validatePauseReplicationRequest(request)
 
                 // Restoring Index can't be paused
-                val restoring = clusterService.state().custom<RestoreInProgress>(RestoreInProgress.TYPE).any { entry ->
+                val restoring = clusterService.state().custom<RestoreInProgress?>(RestoreInProgress.TYPE)?.any { entry ->
                     entry.indices().any { it == request.indexName }
-                }
+                } ?: false
```

Comment:
we have similar code in `transportStopReplication` as well which will need this fix .
btw, why do we need this fix ?

Reply:
We are hitting the issues if entry is not present. Updated the code